com.ca.lsp.cobol/lsp-core-cobol-parser/src/main/java/com/ca/lsp/core/cobol/visitor/CobolVisitor.java

```diff
@@ -64,35 +70,31 @@ public CobolVisitor(
   public Class visitIdentificationDivision(IdentificationDivisionContext ctx) {
     Token token = ctx.getStart();
     areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
-    return visitChildren(ctx);
-  }
-
-  @Override
-  public Class visitProgramIdParagraph(ProgramIdParagraphContext ctx) {
-    if (ctx.programName() != null) {
-      programName = ctx.programName().getText();
-    }
+    outlineTreeBuilder.addNodeToParent("IDENTIFICATION DIVISION", NodeType.DIVISION, ctx);
     return visitChildren(ctx);
   }
 
   @Override
   public Class visitProcedureDivision(ProcedureDivisionContext ctx) {
     Token token = ctx.getStart();
     areaAWarning(ctx.start.getCharPositionInLine(), token.getText(), token.getLine());
+    outlineTreeBuilder.addNodeToParent("PROCEDURE DIVISION", NodeType.DIVISION, ctx);
```

Comment:
These divisions are good candidates to be reused, would be better to define them either const or enum "PROCEDURE DIVISION", "ENVIRONMENT DIVISION", "DATA DIVISION"